src/test/java/teammates/test/cases/testdriver/TestngTest.java

```diff
@@ -0,0 +1,80 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.testng.annotations.Test;
+
+import teammates.common.util.FileHelper;
+import teammates.test.cases.BaseTestCase;
+
+public class TestngTest extends BaseTestCase {
+
+    HashMap<String, String> tests = new HashMap<String, String>(); // <class name, package name>
```

Comment:
This need not be an instance level variable.  Having this as a instance variable is causing some of the bad code below.